src/recorder.js

```diff
@@ -151,10 +151,24 @@ Recorder.prototype.initWorker = function(){
   });
 };
 
-Recorder.prototype.pause = function(){
-  if ( this.state === "recording" ){
+Recorder.prototype.pause = function( flush ) {
+  if ( this.state === "recording" ) {
     this.state = "paused";
+    if ( flush && this.config.streamPages ) {
+      return new Promise((resolve, reject) => {
+        const callback = (e) => {
+          if ( e["data"]["message"] === 'flushed' ) {
+            this.encoder.removeEventListener( "message", callback );
```

Comment:
This seems a bit weird to me. Is it important to have a promise returned from `pause`?
Whats the use case for having both the `onpause` callback as well as a promise. Is it just for testing?

Reply:
It provides some uniformity in the interface - start returns a promise, so it makes sense for the other async methods to do so too (giving users the choice of either using the callbacks, or using the promises). In my case, I wait until paused to get the current encoded sample (very small PR, once this one is done), so I can precisely mark where the pause happened in the stream.

If you want to stick with just the callbacks for onpause and onstop for simplicity, I could update the PR (and my code to match - I'm using the promise API locally at the moment). The tests can use the callback in that case.

Reply:
I don't see any problem to add promises to the APIs. One thing we could improve is the implementation. One binds and unbinds a callback handler, the other uses the global message handler. I prefer the callback handler binding over the global callbacks, as it allows for multiple callbacks to be bound.

Reply:
The concern is that the promise ideally should predictably resolve after onstop (and ondataavailable in the non-streaming case) is called. If I use addEventListener here, we will be relying on the order of the listeners being called matching registration order. I think this is required by the DOM level 3 event spec, but not earlier ones. Given this library requires a reasonably modern browser anyway, it's probably ok to rely on that, so I'll make the change.